src/Cli/dotnet/Program.cs

```diff
@@ -238,17 +238,26 @@ internal static int ProcessArgs(string[] args, TimeSpan startupTime, ITelemetry
             else
             {
                 PerformanceLogEventSource.Log.ExtensibleCommandResolverStart();
-                var resolvedCommand = CommandFactoryUsingResolver.Create(
-                        "dotnet-" + parseResult.GetValue(Parser.DotnetSubCommand),
-                        args.GetSubArguments(),
-                        FrameworkConstants.CommonFrameworks.NetStandardApp15);
-                PerformanceLogEventSource.Log.ExtensibleCommandResolverStop();
+                try
+                {
+                    var resolvedCommand = CommandFactoryUsingResolver.Create(
+                            "dotnet-" + parseResult.GetValue(Parser.DotnetSubCommand),
+                            args.GetSubArguments(),
+                            FrameworkConstants.CommonFrameworks.NetStandardApp15);
+                    PerformanceLogEventSource.Log.ExtensibleCommandResolverStop();
 
-                PerformanceLogEventSource.Log.ExtensibleCommandStart();
-                var result = resolvedCommand.Execute();
-                PerformanceLogEventSource.Log.ExtensibleCommandStop();
+                    PerformanceLogEventSource.Log.ExtensibleCommandStart();
+                    var result = resolvedCommand.Execute();
+                    PerformanceLogEventSource.Log.ExtensibleCommandStop();
 
-                exitCode = result.ExitCode;
+                    exitCode = result.ExitCode;
+                }
+                catch (CommandUnknownException e)
```

Comment:
I'm thinking it'd probably be cleaner to override the ToString method on CommandUnknownException, but this is fine in my opinion.

Reply:
I would agree. However, overriding ToString method cannot change the color of the text. No matter how we override it, the the toString method returns a string. Since CommandUnknownException is an Exception, the message will be red.

Reply:
Good point 🙂